Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/OfferingsList.swift

```diff
@@ -13,67 +13,104 @@ import RevenueCatUI
-                )
-
-                self.offerings = .success(
-                    .init(
-                        sections: Array(offeringsBySection.keys).sorted { $0.description < $1.description },
-                        offeringsBySection: offeringsBySection
+        self.content
+            .navigationTitle("Paywalls")
+            .task {
+                do {
+                    async let appOfferings = fetchOfferings(for: app).all
+                    async let appPaywalls = fetchPaywalls(for: app).all
+                    
+                    let offerings = try await appOfferings
+                    let paywalls = try await appPaywalls
+                    
+                    let offeringPaywallData = OfferingPaywallData(offerings: offerings, paywalls: paywalls)
+                    
+                    self.offeringsPaywalls = .success(
+                        offeringPaywallData.paywallsByOffering()
                     )
-                )
-            } catch let error as NSError {
-                self.offerings = .failure(error)
+                    
+                } catch let error as NSError {
+                    self.offeringsPaywalls = .failure(error)
+                }
+            }
+    }
+    
+    public func fetchOfferings(for app: DeveloperResponse.App) async throws -> OfferingsResponse {
+        return try await self.client.perform(
+            .init(
+                method: .get,
+                endpoint: .offerings(projectID: app.id)
+            )
+        )
+    }
+    
+    public func fetchPaywalls(for app: DeveloperResponse.App) async throws -> PaywallsResponse {
+        return try await self.client.perform(
+            .init(
+                method: .get,
+                endpoint: .paywalls(projectID: app.id)
+            )
+        )
+    }
+    
+    private struct OfferingPaywallData {
+
+        var offerings: [OfferingsResponse.Offering]
+        var paywalls: [PaywallsResponse.Paywall]
+        
+        func paywallsByOffering() -> [OfferingPaywall] {
+            let paywallsByOfferingID = Set(self.paywalls).dictionaryWithKeys { $0.offeringID }
+
+            var offeringPaywall = [OfferingPaywall]()
+            for offering in self.offerings {
+                if let paywall = paywallsByOfferingID[offering.id] {
+                    offeringPaywall.append(OfferingPaywall(offering: offering, paywall: paywall))
+                }
             }
+
+            return offeringPaywall
         }
     }
 
     @ViewBuilder
     private var content: some View {
-        switch self.offerings {
+        switch self.offeringsPaywalls {
         case let .success(data):
             VStack {
                 Text(Self.modesInstructions)
                     .font(.footnote)
-                self.list(with: data)
+                if data.isEmpty {
+                    ContentUnavailableView("No paywalls configured", systemImage: "exclamationmark.triangle.fill")
```

Comment:
oh nice I love that we get to play with `ContentUnavailableView` now that we're targeting 17